charts/nr-k8s-otel-collector/values.yaml

```diff
@@ -91,12 +86,1192 @@ deployment:
   envs: []
   # -- Sets additional environment variable sources for the deployment.
   envsFrom: []
-  # -- Settings for deployment configmap
-  # @default -- See `values.yaml`
-  configMap:
-    # -- OpenTelemetry config for the deployment. If set, overrides default config and disables configuration parameters for the deployment.
-    config: {}
 
+otel:
+  receivers:
+    deployment:
```

Comment:
Not a blocker, 
Looks like we're separating configs into deployment, daemonset and shared. 
When I looked into it, it looks like otel discards configs that aren't applied to a pipeline. 
So we could remove the distinction between configs at this level. 
Would make this file a little easier to read and remove some exists checks in our templating.

Reply:
My preference would be to separate by deployment/daemonset/shared since templating logic in `_collector_config.tpl `could target things easier, currently this is what it does for certain configs under each type